### PR TITLE
First version of cache builder and lookup logic for live validation

### PR DESCRIPTION
**Pull request by vishrutshah**
- [x] Clones git repository from configurations
- [x] Read all the swagger spec files and parse them
- [x] Create an in memory cache for all the operations across all the swagger specs based on below structure:
```json5
{
  "api-version1":
  {
    "provider1": 
     {
        "get": [ {opeationObject(Sway)}, {operationObject(sway)}, . .  ],
        "post": [ {opeationObject(Sway)}, {operationObject(sway)}, . .  ],
        "put": [ {opeationObject(Sway)}, {operationObject(sway)}, . .  ],
        ...
     }
  },
  "api-version2":
  {
    "provider2": 
     {
        "delete": [ {opeationObject(Sway)}, {operationObject(sway)}, . .  ],
        "head": [ {opeationObject(Sway)}, {operationObject(sway)}, . .  ],
        "get": [ {opeationObject(Sway)}, {operationObject(sway)}, . .  ],
        ...
     }
  },
  . . .
}
```

- [x] Expose `getPotentialOperations` for lookup for best match on regex of path

Part of https://github.com/Azure/openapi-validation-tools/issues/89 & https://github.com/Azure/openapi-validation-tools/issues/90

### CODE REVIEW

**Comment by vishrutshah** [comment]
Closing and reopening to run travis

**Comment by amarzavery** [comment]
@vishrutshah  **શું એક અદ્ભુત કામ ડ્યૂડ. તમે પ્રથમ પુલ વિનંતી ટ્રેવિસ CI સક્રિય થયેલ**

File: test/liveValidatorTests.js [hunk 1]
```diff
@@ -0,0 +1,176 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+var assert = require('assert');
+var LiveValidator = require('../lib/liveValidator.js');
+
+describe('Live Validator', function () {
+    describe('Initialization', function () {
```
Comment (amarzavery) [inline test/liveValidatorTests.js]: indentation 2 spaces

Comment (vishrutshah) [inline test/liveValidatorTests.js]: sure i'll update my vscode config for 2 space indentations 👍

**Pull request merged**